front-end/generate-task-lists.js

```diff
@@ -1,127 +1,167 @@
+        .forEach(task => {
+            const currentTask = document.createElement('li');
+            const buttonIsDone = document.createElement('button');
+            const buttonIsDeleted = document.createElement('button');
+            const taskContent = document.createElement('span');
+            const taskDeadlineTime = document.createElement('time');
+
+            currentTask.classList.add('task-list__item');
+            buttonIsDone.classList.add(
+                'task-list-item__button',
+                'task-list-item__button--is-done'
+            );
+            buttonIsDeleted.classList.add(
+                'task-list-item__button',
+                'task-list-item__button--is-deleted'
+            );
+            taskContent.classList.add('task-list-item__content');
+            taskDeadlineTime.classList.add('task-list-item__deadline');
+
+            taskContent.innerText = task.content;
+            taskDeadlineTime.innerText = moment(task.deadline).format(
+                'YYYY.MM.DD'
+            );
+
+            buttonIsDone.innerHTML = '&#10003;';
+            buttonIsDeleted.innerHTML = '&#10060;';
+
+            currentTask.appendChild(buttonIsDone);
+            currentTask.appendChild(taskContent);
+            currentTask.appendChild(taskDeadlineTime);
+            currentTask.appendChild(buttonIsDeleted);
+
+            taskList.appendChild(currentTask);
+
+            document.querySelector('.task-form').reset();
+
+            const markTaskAsDone = event => {
+                task.status = STATUSES.DONE;
+                renderDoneList();
+                renderTaskList();
+            };
+            const markTaskAsDelete = event => {
+                task.status = STATUSES.DELETED;
+                renderDeletedList();
+                renderTaskList();
+            };
+
+            buttonIsDone.addEventListener('click', markTaskAsDone);
+            buttonIsDeleted.addEventListener('click', markTaskAsDelete);
+        });
+};
 
 const addTask = event => {
-	event.preventDefault()
-
-	const taskName = document.querySelector('.task-form__input--task-content').value
-	const taskDeadline = document.querySelector('.task-form__input--date').value
-
-	tasks.push({
-		id: uuid(),
-		name: taskName,
-		deadline: taskDeadline,
-		status: STATUSES.NEW,
-	})
-	renderTaskList()
-}
-
-addTaskForm.addEventListener('submit', addTask)
-
-const taskListDone = document.querySelector('.task-list__performed-activities--is-done')
-const taskListDeleted = document.querySelector('.task-list__performed-activities--is-deleted')
+    event.preventDefault();
+
+    const taskContent = document.querySelector(
+        '.task-form__input--task-content'
+    ).value;
+    const taskDeadline = document.querySelector('.task-form__input--date')
+        .value;
+
+    return axios
+        .post('/task', {
+            content: taskContent,
+            deadline: taskDeadline,
+            status: STATUSES.NEW,
```

Comment:
No need to send a `status` field at all, it's "NEW" by default, mongoose provides that

Reply:
I've unresolved the comment because you made no change